eclair-core/src/main/scala/fr/acinq/eclair/wire/protocol/OfferCodecs.scala

```diff
@@ -0,0 +1,249 @@
+
+  private val currency: Codec[Currency] = variableSizeBytesLong(varintoverflow, utf8).as[Currency]
+
+  private val amount: Codec[Amount] = variableSizeBytesLong(varintoverflow, tmillisatoshi).as[Amount]
+
+  private val description: Codec[Description] = variableSizeBytesLong(varintoverflow, utf8).as[Description]
+
+  private val features: Codec[FeaturesTlv] = variableSizeBytesLong(varintoverflow, bytes).xmap[Features[FeatureScope]](Features(_), _.toByteVector).as[FeaturesTlv]
+
+  private val absoluteExpiry: Codec[AbsoluteExpiry] = variableSizeBytesLong(varintoverflow, tu64overflow).as[TimestampSecond].as[AbsoluteExpiry]
+
+  private val hopCodec: Codec[BlindedNode] = (("nodeId" | publicKey) :: ("encTlv" | variableSizeBytes(uint16, bytes))).as[BlindedNode]
+
+  private val pathCodec: Codec[BlindedRoute] = (("firstNodeId" | publicKey) :: ("blinding" | publicKey) :: ("path" | listOfN(uint8, hopCodec).xmap[Seq[BlindedNode]](_.toSeq, _.toList))).as[BlindedRoute]
+
+  private val paths: Codec[Paths] = variableSizeBytesLong(varintoverflow, list(pathCodec)).xmap[Seq[BlindedRoute]](_.toSeq, _.toList).as[Paths]
+
+  private val issuer: Codec[Issuer] = variableSizeBytesLong(varintoverflow, utf8).as[Issuer]
+
+  private val quantityMin: Codec[QuantityMin] = variableSizeBytesLong(varintoverflow, tu64overflow).as[QuantityMin]
+
+  private val quantityMax: Codec[QuantityMax] = variableSizeBytesLong(varintoverflow, tu64overflow).as[QuantityMax]
+
+  // Only the x component of the key is serialized. We arbitrarily add hex"02" to get a valid key.
+  // When using this key, we need to try both the version starting with hex"02" and the one starting with hex"03".
+  private val nodeId: Codec[NodeId] = variableSizeBytesLong(varintoverflow, bytes32).xmap[PublicKey](b32 => PublicKey(hex"02" ++ b32), key => ByteVector32(key.value.drop(1))).as[NodeId]
+
+  private val sendInvoice: Codec[SendInvoice] = variableSizeBytesLong(varintoverflow, provide(SendInvoice()))
+
+  private val refundFor: Codec[RefundFor] = variableSizeBytesLong(varintoverflow, bytes32).as[RefundFor]
+
+  private val signature: Codec[Signature] = variableSizeBytesLong(varintoverflow, bytes64).as[Signature]
+
+  val offerTlvCodec: Codec[TlvStream[OfferTlv]] = TlvCodecs.tlvStream[OfferTlv](discriminated[OfferTlv].by(varint)
+    .typecase(UInt64(2), chains)
+    .typecase(UInt64(6), currency)
+    .typecase(UInt64(8), amount)
+    .typecase(UInt64(10), description)
+    .typecase(UInt64(12), features)
+    .typecase(UInt64(14), absoluteExpiry)
+    .typecase(UInt64(16), paths)
+    .typecase(UInt64(20), issuer)
+    .typecase(UInt64(22), quantityMin)
+    .typecase(UInt64(24), quantityMax)
+    .typecase(UInt64(30), nodeId)
+    .typecase(UInt64(54), sendInvoice)
+    .typecase(UInt64(34), refundFor)
+    .typecase(UInt64(240), signature)).complete
+
+  val offerCodec: Codec[Offer] = offerTlvCodec.narrow({ tlvs =>
+    if (tlvs.get[Description].isEmpty) {
```

Comment:
Right, we want to keep offers as small as possible, so it makes sense to omit the signature because we will sign invoices, which should be enough for the payer/caller. It's interesting that it allows anyone to create an offer for your node, which can be a feature (maybe?).